kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -273,4 +276,38 @@ private Protocol getInitialProtocol() {
                 null /* readerFeatures */,
                 null /* writerFeatures */);
     }
+
+    /**
+     * Utility methods adds the test table properties to the given properties. This helps in
+     * repeating a single test suite with different table properties.
+     * <p>
+     * E.g. we have a test suite for inserting into a non-column mapping mode. We can repeat the
+     * same tests in a new suite with column mapping mode enabled by adding column mapping related
+     * properties for every transaction that is built.
+     * <p>
+     * It relies on the Java asserts to ensure that it is not called in production and production
+     * code is not affected by this method.
+     * <p>
+     */
+    private void addTestTableProperties() {
+        boolean assertsEnabled = false;
+        assert assertsEnabled = true; // Value will be set to true when Java asserts are enabled.
```

Comment:
the `assert` by default doesn't do anything and needs to be explicitly enabled via `-ea`(which most projects don't do). I actually never saw a project that enabled assertions in production code, simply because a user can't do anything to revover the application if such an assert fails

Reply:
Please see the comments. This is just for tests. This helps avoid repeating the same tests for each column mapping mode.